driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -349,9 +349,10 @@ public ConnectionString(final String connectionString) {
 
         Map<String, List<String>> connectionStringOptionsMap = parseOptions(connectionStringQueryParamenters);
         Map<String, List<String>> txtRecordsOptionsMap = parseOptions(txtRecordsQueryParameters);
-        translateOptions(combineOptionsMaps(txtRecordsOptionsMap, connectionStringOptionsMap));
-        credential = createCredentials(connectionStringOptionsMap, userName, password);
-        warnOnUnsupportedOptions(connectionStringOptionsMap);
+        Map<String, List<String>> combinedOptionsMaps = combineOptionsMaps(txtRecordsOptionsMap, connectionStringOptionsMap);
+        translateOptions(combinedOptionsMaps);
+        credential = createCredentials(combinedOptionsMaps, userName, password);
```

Comment:
No way to test this change currently because none of the TXT records have auth information in them (e.g a authSource)